client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts

```diff
@@ -141,48 +150,21 @@ const siteMigration: Flow = {
 					};
 
 					if ( action === 'skip_platform_identification' || platform !== 'wordpress' ) {
-						if ( isHostedSiteMigrationFlow( variantSlug ?? '' ) ) {
-							// siteId/siteSlug wont be defined here if coming from a direct link/signup.
-							// We need to make sure there's a site to import into.
-							if ( ! siteSlugParam ) {
-								return navigate(
-									addQueryArgs( { from, skipMigration: true }, STEPS.SITE_CREATION_STEP.slug )
-								);
-							}
-						}
-						return exitFlow(
-							addQueryArgs(
-								{
-									siteId,
-									siteSlug,
-									from,
-									origin: STEPS.SITE_MIGRATION_IDENTIFY.slug,
-									backToFlow: `/${ flowPath }/${ STEPS.SITE_MIGRATION_IDENTIFY.slug }`,
-								},
-								'/setup/site-setup/importList'
-							)
+						return navigate(
+							addQueryArgs( { from, skipMigration: true }, STEPS.SITE_CREATION_STEP.slug )
 						);
 					}
 
-					if ( isHostedSiteMigrationFlow( variantSlug ?? '' ) ) {
-						if ( ! siteSlugParam ) {
-							if ( siteCount > 0 ) {
-								return navigate( `sitePicker?from=${ encodeURIComponent( from ) }` );
-							}
-
-							if ( from ) {
-								return navigate( addQueryArgs( { from }, STEPS.SITE_CREATION_STEP.slug ) );
-							}
+					if ( ! hasPreloadedSite && userHasWpComSites ) {
+						return navigate( addQueryArgs( { from }, STEPS.PICK_SITE.slug ) );
+					}
```

Comment:
From a logic perspective, the sequence here doesn't feel right. I think we should do the following:
 * If the user doesn't have any existing sites, we should create a new site
 * If the user has specified a site, we should use that site -- we shouldn't create a new one
 * If the user has existing sites, but hasn't specified a site, we should show the site picker